File: Easy/Seating Arrangement/seating-arrangement.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        int test = sc.nextInt();
        while(test-- > 0){
            int n = sc.nextInt();
            int m = sc.nextInt();
            int [] seats = new int[m];
            for(int i = 0; i < m; i++){
                seats[i] = sc.nextInt();
            }
            Solution obj = new Solution();
            boolean res = obj.is_possible_to_get_seats(n, m, seats);
            
            String _result_val = (res) ? "Yes" : "No";
            System.out.println(_result_val);
        }
    }
}

// } Driver Code Ends


public class Solution {
    public static boolean is_possible_to_get_seats(int n, int m, int[] seats) {
        for(int i = 0; i < m; i++) {
            if(seats[i] == 1) {
                if(i - 1 >= 0) {
                    seats[i - 1] = 2;
                }
                if(i + 1 < m && seats[i + 1] == 0) {
                    seats[i + 1] = 2;
                }
            }
        }
        int cnt = 0;
        for(int i = 0; i < m; i++) {
            if(seats[i] == 0) {
                cnt++;
                if(i - 1 >= 0) {
                    seats[i - 1] = 2;
                }
                if(i + 1 < m) {
                    seats[i + 1] = 2;
                }
            }
        }
        return cnt >= n;
    }
}

        

File: Easy/Seating Arrangement/seating-arrangement.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++
//User function Template for C++

class Solution{
    public:
    bool is_possible_to_get_seats(int n, int m, vector<int>& seats){
        // Write your code here.
        vector<int>temp = seats;
        int cnt = 0;
        if(temp[0]==0 && temp[1]==0) temp[0] = 1,cnt++;
        for(int i=1;i<m-1;i++)
            if(temp[i-1]==0 && temp[i+1]==0 && temp[i]==0)
                temp[i]=1,cnt++;
        if(temp[m-1]==0 && temp[m-2]==0) cnt++;
        return cnt>=n ? 1:0;
    }
};

//{ Driver Code Starts.

int main(){

    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int m;
        cin >> m;
        vector<int> seats(m);
        for (int i = 0; i < m; i++) {
            cin >> seats[i];
        }
        Solution obj;
        if (obj.is_possible_to_get_seats(n, m, seats)) {
            cout << "Yes" << endl;
        }
        else {
            cout << "No" << endl;
        }
    }
}

// } Driver Code Ends